o1vm/src/pickles/prover.rs

```diff
@@ -176,7 +190,72 @@ where
         }
     };
 
-    // FIXME: add quotient polynomial
+    let quotient_poly: DensePolynomial<G::ScalarField> = {
+        let mut last_constraint_failed = None;
+        // Only for debugging purposes
+        for expr in constraints.iter() {
+            let fail_q_division =
+                ProverError::ConstraintNotSatisfied(format!("Unsatisfied expression: {:}", expr));
+            // Check this expression are witness satisfied
+            let (_, res) = expr
+                .evaluations(&column_env)
+                .interpolate_by_ref()
+                .divide_by_vanishing_poly(domain.d1)
```

Comment:
are we paying for a division for each constraints just to check, and re doing that division line 246 ?

Reply:
Yes, I am adding an option soon to activate only this in a "debug" mode. PR coming.

Reply:
I'd remove that, doesn't it pollute the code ?
Note that we can check the reminder when dividing by the vanishing poly if we really want a check

Reply:
I did remove https://github.com/o1-labs/proof-systems/pull/2652/commits/8e87244a98d55b90d175ad389611a3c98bd16b34 and https://github.com/o1-labs/proof-systems/pull/2652/commits/96d42c127ef025869c91e5fed680e0e383108706 as you seem to have a strong opinion on this. I would probably keep the commits somewhere on my side for debugging purposes when finishing the prover.